OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -81,5 +90,14 @@ WeaponInfo ChooseWeaponForExplosion(Actor self)
 			var useFullExplosion = self.World.SharedRandom.Next(100) <= info.LoadedChance;
 			return (shouldExplode && useFullExplosion) ? info.WeaponInfo : info.EmptyWeaponInfo;
 		}
+
+		public void Damaged(Actor self, AttackInfo e)
+		{
+			if (info.HealthPercentage == 0)
+				return;
+
+			if ((((float)health.HP / (float)health.MaxHP) * 100) < 100 - info.HealthPercentage)
```

Comment:
You can't use floating point calculations anywhere in simulation code.  This leads to non-deterministic behaviour and desyncs.